cmd/pdc/main.go

```diff
@@ -2,21 +2,165 @@ package main
 
 import (
 	"context"
+	"flag"
+	"fmt"
 	"os"
 	"os/signal"
 	"syscall"
 
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
 	"github.com/grafana/dskit/services"
+	"github.com/grafana/pdc-agent/pkg/pdc"
 	"github.com/grafana/pdc-agent/pkg/ssh"
 )
 
+type mainFlags struct {
+	PrintHelp bool
+	LogLevel  string
+}
+
+func (mf *mainFlags) RegisterFlags(fs *flag.FlagSet) {
+	fs.BoolVar(&mf.PrintHelp, "h", false, "Print help")
+	fs.StringVar(&mf.LogLevel, "log.level", "info", `"debug", "info", "warn" or "error"`)
+}
+
 func main() {
+	sshConfig := ssh.DefaultConfig()
+	mf := &mainFlags{}
+	pdcClientCfg := &pdc.Config{}
+
+	sshConfig.Args = os.Args[1:]
+
+	legacyMode := inLegacyMode()
+
+	if legacyMode {
+		sshConfig.LegacyMode = true
+		err := runLegacyMode(sshConfig)
+		if err != nil {
+			fmt.Printf("error: %s", err)
+			os.Exit(1)
+		}
+		return
+	}
+
+	usageFn, err := parseFlags(mf.RegisterFlags, sshConfig.RegisterFlags, pdcClientCfg.RegisterFlags)
+	if err != nil {
+		fmt.Println("cannot parse flags")
+		os.Exit(1)
+	}
+
+	logger := setupLogger(mf.LogLevel)
+
+	if mf.PrintHelp {
+		usageFn()
+		return
+	}
+
+	err = run(logger, sshConfig, pdcClientCfg)
+	if err != nil {
+		level.Error(logger).Log("err", err)
+		os.Exit(1)
+	}
+
+	sshConfig.PDC = *pdcClientCfg
```

Comment:
This should be moved above the run command

Reply:
doh I had this locally already